[docs] clarify `expo.version` vs `expo.ios.buildNumber`

cruzach opened the pull request:
# Why

Closes #4732

cruzach committed:
docs: clarify expo.version vs expo.ios.buildnumber

cruzach committed:
SDK 33 docs

cruzach committed:
SDK 32

brentvatne commented:
this looks good! i'd also add some information to the [Deploying to App Stores](https://docs.expo.io/versions/v33.0.0/distribution/app-stores/) section. maybe we need a heading with more information about versions, so pull out "Use versionCode and buildNumber to distinguish different binaries of your app." from [here](https://docs.expo.io/versions/v33.0.0/distribution/app-stores/#tailor-your-app-metadata) and put it under a new section where we go into more detail on the options and what people need to consider when changing version numbers

cruzach committed:
add `versioning` section

explains the different fields in `app.json` that correspond with Android and iOS versioning

cruzach commented:
Done, let me know if you think that's a good amount of detail. Could link to to the native docs for each field, but we already do that in the `configuration with app.json` page (which we link to anyways)

ide commented on docs/pages/versions/v33.0.0/distribution/app-stores.md:
```diff
@@ -31,7 +31,14 @@ Try your app on tablets in addition to handsets. Even if you have `ios.supportsT
+
+You'll use the `app.json` file to specify the version of your app, but there are a few different fields each with specific functionality.
+
+- [`version`](../../workflow/configuration/#version) will apply both to iOS and Android. For iOS, this corresponds to `CFBundleShortVersionString`, and for Android this corresponds to `versionName`. This is your user-facing version string for both platforms.
```
What is it on web?

sergeichestakov replied:
[Null](https://github.com/expo/expo/blob/master/packages/expo-constants/src/ExponentConstants.web.ts#L71). I added this as part of https://github.com/expo/expo/pull/3894

ide commented:
We should also document `nativeAppVersion` and `nativeBuildVersion` (fields of expo-constants). Those are for reading (as opposed to setting) the versions, and are universal ways to access the short version (CFBundleShortVersionString) and non-short version (CFBundleVersion) strings, respectively, across Android and iOS.

cruzach committed:
update app-stores.md unversioned

cruzach committed:
update app-stores.md v32
